libs/platform/src/lib/domain/data-model.ts

```diff
@@ -27,3 +27,30 @@ export function isSelectItem(item: SelectItem): item is SelectItem {
     return item && item.label !== undefined && item.value !== undefined;
 }
 
+/**
+ * To unify the work with domain objects we have these set of interfaces that each Entity or Value
+ * must use to leverage some of the functionality we have in the core
+ *
+ */
+export interface CompositeType {
+
+  className(): string;
+}
+export interface Deserializable {
+  getTypes(): any;
+}
+
+/**
+ * EntityComposite having identity that can be identified in the storage by its ID. Entities are
+ * mutable objects
+ */
+export interface Entity extends Identity,
+  Deserializable,
+  CompositeType {
+}
+
```

Comment:
Missing Typedocs

Reply:
> Missing Typedocs

already in place.


<img width="849" alt="Screenshot 2020-04-17 at 1 52 38 PM" src="https://user-images.githubusercontent.com/53534379/79548244-c708d400-80b2-11ea-9b68-ef830c468791.png">